agreementbot/consumer_protocol_handler.go

```diff
@@ -378,7 +378,7 @@ func (b *BaseConsumerProtocolHandler) HandlePolicyChangeForAgreement(ag persiste
 
 	// cancel the agreement if the agreement is not finalized yet
 	if ag.AgreementFinalizedTime == 0 {
-		return false, false
+		return true, true
```

Comment:
I don't think this is correct.... I think this 
```
// cancel the agreement if the agreement is not finalized yet
	if ag.AgreementFinalizedTime == 0 {
		return true, true
}	
```

should be moved after line 436 that checks the match... Only if there isn't a match, then shouldn't the agreement be canceled here?  I was thinking only if there is still a match but the agreement is not finalized, that only the agreement update would not occur... That is what I ran w/ yesterday